Sources/SKSupport/AsyncUtils.swift

```diff
@@ -65,3 +65,27 @@ public func withCancellableCheckedThrowingContinuation<Handle, Result>(
     onCancel: callCancel
   )
 }
+
+extension Collection {
+  /// Transforms all elements in the collection concurrently and returns the transformed collection.
+  public func concurrentMap<TransformedElement>(
+    _ transform: @escaping (
+      Element
+    ) async -> TransformedElement
+  ) async -> [TransformedElement] {
+    return await withTaskGroup(of: (Int, TransformedElement).self) { taskGroup in
```

Comment:
I'd recommend limiting this to some bounded number of tasks, otherwise your tasks may start thrashing the scheduler

Reply:
Yeah have you considered doing something like this instead. 
https://github.com/adam-fowler/async-collections/blob/cf064405329667bbeb66d55563bbf14048faed2c/Sources/AsyncCollections/Map.swift#L90

It adds a limit to the number of tasks that can be spawn and removes the requirement for `sort`

Reply:
Good idea 👍🏽